response_operations_ui/controllers/survey_controllers.py

```diff
@@ -137,3 +138,17 @@ def create_survey(survey_ref, short_name, long_name, legal_basis):
         raise ApiError(response)
 
     logger.debug('Successfully created new survey', survey_ref=survey_ref)
+
+
+def get_survey_by_id(survey_id):
+    logger.debug("Retrieve survey using survey id", survey_id=survey_id)
+    url = f'{app.config["SURVEY_URL"]}/surveys/{survey_id}'
+    response = requests.get(url, auth=app.config['SURVEY_AUTH'])
+
+    try:
+        response.raise_for_status()
+    except (HTTPError, RequestException):
+        logger.exception("Survey retrieval failed", survey_id=survey_id)
```

Comment:
Similar to the other ones, can respond with a 404 which we don't want to log at error level